.github/workflows/main.yml

```diff
@@ -66,18 +66,54 @@ jobs:
           ninja
           ninja test
 
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+
+      - name: Setup minio
+        run: |
+          docker run -d -p 9000:9000 --name minio \
+                     -e "MINIO_ACCESS_KEY=minioadmin" \
+                     -e "MINIO_SECRET_KEY=minioadmin" \
+                     -v /tmp/data:/data \
+                     -v /tmp/config:/root/.minio \
+                     minio/minio server /data
+
+          sudo docker start minio
+
+          export AWS_ACCESS_KEY_ID=minioadmin
+          export AWS_SECRET_ACCESS_KEY=minioadmin
+          export AWS_EC2_METADATA_DISABLED=true
+
+          # pip3 install awscli --force-reinstall --upgrade
+
+
       - name: powerloader python based tests
         shell: bash -l -eo pipefail {0}
         if: runner.os != 'Windows'
         env:
+          LOCAL_AWS_S3_BUCKET: testbucket
+          LOCAL_AWS_S3_ACCESS_KEY: minioadmin
+          LOCAL_AWS_S3_SECRET_KEY: minioadmin
+          LOCAL_AWS_S3_ENDPOINT: http://127.0.0.1:9000
+
           AWS_ACCESS_KEY: ${{ secrets.AWS_ACCESS_KEY }}
           AWS_SECRET_KEY: ${{ secrets.AWS_SECRET_KEY }}
           AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
         run: |
-          # There is probably a better way of doing this...
+          # Download the packages from conda forge
           server_path="test/conda_mock/static/packages"
           powerloader_path="build/powerloader"
           wget https://beta.mamba.pm/get/conda-forge/osx-arm64/xtensor-0.24.0-hc021e02_0.tar.bz2 \
               -O ./test/conda_mock/static/packages/xtensor-0.24.0-hc021e02_0.tar.bz2
           $powerloader_path download -f test/remote_mirrors.yml -d $server_path
+
+          # Populate minIO
+          # https://docs.min.io/docs/aws-cli-with-minio.html
+          aws --endpoint-url $LOCAL_AWS_S3_ENDPOINT s3 mb s3://testbucket
```

Comment:
this doesn't work if you don't install the `awscli`

Reply:
I installed it manually on my machine and then added it to `environment.yml`

Reply:
makes sense